extras/shader-nodes/texSample.js

```diff
@@ -0,0 +1,24 @@
+// node
+var texSample = {
+    code: "void texSample(in sampler2D tex, in vec2 uv, out vec4 rgba, out vec3 rgb, out float r, out float g, out float b, out float a) \n{\n    rgba = texture2D(tex, uv);\n    rgb = rgba.rgb;\n    r = rgba.r;\n    g = rgba.g;\n    b = rgba.b;\n    a = rgba.a;\n}",
+    meta: { label: "TEXTURE SAMPLE" }
+};
+
+// ports
+texSample.meta.ports = [];
+texSample.meta.ports[0] = { label: 'TEXTURE', id: 0 };
+texSample.meta.ports[1] = { label: 'UV', id: 1 };
+texSample.meta.ports[2] = { label: 'RGBA', id: 2 };
+texSample.meta.ports[3] = { label: 'RGB', id: 3 };
+texSample.meta.ports[4] = { label: 'R', id: 4 };
+texSample.meta.ports[5] = { label: 'G', id: 5 };
+texSample.meta.ports[6] = { label: 'B', id: 6 };
+texSample.meta.ports[7] = { label: 'A', id: 7 };
```

Comment:
write these in mode condensed form I think. Example:

```
var texSample = {
code = "something",
meta = {
    label: "blah",
    ports = [
         { label: 'TEXTURE', id: 0 },
         { label: 'UV', id: 1 
    ]
    }
}
```

Reply:
even one more idea would be to separate meta data that are needed for shader generation from metadata only used in editor .. so that at a later stage we can exclude them from the file. 
```
var texSample = {
code = "something",
meta = {
    ports = [.  // maybe not needed if only ids?
         { id: 0 },
         { id: 1 
    ]
    },
editor = {
    label: "blah",
    ports = [
         { label: 'TEXTURE'},
         { label: 'UV'}
    ]
}
```

Reply:
Thanks for this - I will tidy this up soon - before implementing any more core shaders nodes

Reply:
Tidied up - split up editor data from other data (explicit port ids) - should be possible to split editor data into `playcanvas-shader-nodes-editor.js` later?

Reply:
I think add generator function directly inside, vs having to have separate cross.gen block.

I thought it could be easier to build two js files from these, and in one simply strip the editor part .. similarly to how we strip spacing and comments from shaders and code.

Either way, should be doable.